pkg/scheduler/core/assignment.go

```diff
@@ -54,6 +54,9 @@ type assignState struct {
 func newAssignState(candidates []*clusterv1alpha1.Cluster, strategy *policyv1alpha1.ReplicaSchedulingStrategy, obj *workv1alpha2.ResourceBindingSpec) *assignState {
 	var strategyType string
 
+	if strategy == nil {
+		strategy = &policyv1alpha1.ReplicaSchedulingStrategy{ReplicaSchedulingType: DuplicatedStrategy}
+	}
```

Comment:
Hi @Garrybest 

Do you remember why we introduce `Duplicated` for `replicaSchedulingType`?
https://github.com/karmada-io/karmada/blob/a14d64f055e1011cd774a76d607c4696fb75bc87/pkg/apis/policy/v1alpha1/propagation_types.go#L300-L308

I don't know when people will set the replicaSchedulingType with `Duplicated`, can you remind me?

Reply:
If `Duplicated` is set, a workload must be dispatched to all member clusters. The effect is as same as that `strategy` is set to `nil`. 

BTW, I don't think this design is decent, now we have a lot of if-else judgement in many places. I would like to add a bool field to replace `ReplicaSchedulingType` to represent whether we should divide replicas.

Reply:
Yeah, I'm thinking if we can make some improvements from the API.

Now I suppose if `ReplicaSchedulingStrategy` is set means we are going to `split` replicas, conversely(not set) means we are going to `delicates` replicas. Am I right?

If so, actually we can remove the `ReplicaSchedulingType` from `ReplicaSchedulingStrategy` I think. But we should consider giving `ReplicaSchedulingStrategy` a more clear name.

Reply:
Exactly, just as I thought.

Reply:
Given the fact that we probably not going to promote the API version for a single field of optimization, like this.

I think we need a place to address the changes gonna make for the next version. My idea is to set up the next version API in the codebase, like `pkg/apis/policy/v1alpha2`, but we don't switch to a newer version until enough optimizations are accumulated. What do you think?

Reply:
It's great. We'll switch the API when the time is right.

Reply:
OK. I can do that. And will talk about it at the community meeting before promoting it.